hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/DatanodeStateMachine.java

```diff
@@ -704,6 +711,44 @@ private Thread getCommandHandlerThread(Runnable processCommandQueue) {
     return handlerThread;
   }
 
+  /**
+   * Stops the DiskBalancerService if it is running.
+   */
+  public void stopDiskBalancer() {
+    OzoneContainer ozoneContainer = getContainer();
+    if (ozoneContainer != null) {
+      DiskBalancerService diskBalancerService = ozoneContainer.getDiskBalancerService();
+      if (diskBalancerService != null) {
+        shouldRun = diskBalancerService.getDiskBalancerInfo().isShouldRun();
+        paused.set(true); // Mark as paused
```

Comment:
Based on above, we could combine `shouldRun` and `paused` here by just setting `paused` to `diskBalancerService.getDiskBalancerInfo().isShouldRun()`. It makes the code simpler and still keeps track of whether DiskBalancer was running before pausing.